### PR TITLE
Fix the parameter of TextureCube::setImage, support to pass nullptr to it, keep the same logic as which in web

### PR DESCRIPTION
**Pull request by dumganhar**
-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: native/cocos/core/assets/TextureCube.h [hunk 1]
```diff
@@ -203,7 +203,7 @@ class TextureCube final : public SimpleTexture {
         return _mipmaps.empty() ? nullptr : &_mipmaps[0];
     }
 
-    void setImage(const ITextureCubeMipmap &value);
+    void setImage(const ITextureCubeMipmap *value);
```
Comment (dumganhar) [inline native/cocos/core/assets/TextureCube.h:206]: Getter `TextureCube::getImage` returns `const ITextureCubeMipmap*`, setter needs to pass in pointer too.
It's the same logic as which in web and make swig binding work.
WeChat78f069d578aac33ee7f3b5594c9d3bb6

File: native/cocos/core/assets/TextureCube.h [hunk 2]
```diff
@@ -249,18 +249,18 @@ class TextureCube final : public SimpleTexture {
...
-    ccstd::vector<ITextureCubeMipmap> _mipmaps;
+    bool isRGBE{false};
 
+private:
     /*@serializable*/
-    MipmapMode _mipmapMode{MipmapMode::NONE};
+    ccstd::vector<ITextureCubeMipmap> _mipmaps;
 
     /*@serializable*/
     TextureCubeMipmapAtlasInfo _mipmapAtlas;
```
Comment (dumganhar) [inline native/cocos/core/assets/TextureCube.h:262]: Keep _mipmaps & _mipmapAtlas private

File: native/cocos/core/assets/TextureCube.cpp [hunk 3]
```diff
@@ -168,10 +168,14 @@ void TextureCube::setMipmapAtlasForJS(const TextureCubeMipmapAtlasInfo &value) {
...
 
-void TextureCube::setImage(const ITextureCubeMipmap &value) {
-    _mipmaps.clear();
-    _mipmaps.emplace_back(value);
+void TextureCube::setImage(const ITextureCubeMipmap *value) {
+    if (value != nullptr) {
+        setMipmaps({*value});
+    } else {
+        setMipmaps({});
+    }
```
Comment (dumganhar) [inline native/cocos/core/assets/TextureCube.cpp:176]: WeChatdc3d0a6f9137ff680fe6ef24d88e3814

**Review by minggo — APPROVED** [review]

File: cocos/core/assets/texture-cube.jsb.ts [hunk 4]
```diff
@@ -200,7 +204,7 @@ textureCubeProto._deserialize = function (serializedData: ITextureCubeSerializeD
     const data = serializedData;
     jsb.TextureBase.prototype._deserialize.call(this, data.base, handle);
     this.isRGBE = data.rgbe;
-    this._mipmapMode = parseInt(data.mipmapMode);
+    this._mipmapMode = data.mipmapMode;
```
Comment (dumganhar) [inline cocos/core/assets/texture-cube.jsb.ts]: data.mipmapMode is number, no need to parseInt.
Web implementation doesn't have parseInt logic too.

**Comment by dumganhar** [comment]
@cocos-robot run test cases

**Comment by dumganhar** [comment]
@cocos-robot run test cases

**Pull request merged**